rooms.js

```diff
@@ -439,6 +439,25 @@ class GlobalRoom {
 		}
 		return this.formatList;
 	}
+	get configRankList() {
+		// putting the resultant object in Config would enable this to be run again should config.js be reloaded.
+		if (Config.rankList) {
+			return Config.rankList;
+		}
+		let rankList = [];
+
+		for (let rank in Config.groups) {
+			if (!Config.groups[rank] || !rank) continue;
+
+			let tarGroup = Config.groups[rank];
+			let groupId = tarGroup.addhtml || (!tarGroup.mute && !tarGroup.root) ? 0 : (tarGroup.root || tarGroup.declare) ? 2 : 1;
+
+			rankList.push(rank + ',' + (Config.groups[rank].name || ' ') + ',' + groupId); // send the first character in the rank, incase they put a string several characters long
+		}
+
+		Config.rankList = '|groups|' + rankList.join("|");
+		return Config.rankList;
```

Comment:
You could add `\n` here.

And also check for `Config.nocustomgroulist` here (and return `''` if so).

That would make `onConnect` simpler.